src/bulk/common.ts

```diff
@@ -477,45 +499,45 @@ function mergeBatchResults(batch: any, bulkResult: any, err: any, result: any) {
       }
 
       // Current OpTime TS
-      const opTimeTS = typeof opTime.ts === 'number' ? Long.fromNumber(opTime.ts) : opTime.ts;
-      const opTimeT = typeof opTime.t === 'number' ? Long.fromNumber(opTime.t) : opTime.t;
+      const opTimeTS = typeof opTime?.ts === 'number' ? Long.fromNumber(opTime?.ts) : opTime?.ts;
+      const opTimeT = typeof opTime?.t === 'number' ? Long.fromNumber(opTime?.t) : opTime?.t;
 
       // Compare the opTime's
       if (bulkResult.lastOp == null) {
         bulkResult.lastOp = opTime;
-      } else if (opTimeTS.greaterThan(lastOpTS)) {
+      } else if (lastOpTS && opTimeTS?.greaterThan(lastOpTS)) {
         bulkResult.lastOp = opTime;
-      } else if (opTimeTS.equals(lastOpTS)) {
-        if (opTimeT.greaterThan(lastOpT)) {
+      } else if (lastOpTS && opTimeTS?.equals(lastOpTS)) {
+        if (lastOpT && opTimeT?.greaterThan(lastOpT)) {
           bulkResult.lastOp = opTime;
         }
       }
     }
   }
 
   // If we have an insert Batch type
-  if (batch.batchType === INSERT && result.n) {
+  if (batch.batchType === BatchType.INSERT && result?.n) {
     bulkResult.nInserted = bulkResult.nInserted + result.n;
   }
 
   // If we have an insert Batch type
-  if (batch.batchType === REMOVE && result.n) {
+  if (batch.batchType === BatchType.REMOVE && result?.n) {
     bulkResult.nRemoved = bulkResult.nRemoved + result.n;
   }
 
   let nUpserted = 0;
 
   // We have an array of upserted values, we need to rewrite the indexes
-  if (Array.isArray(result.upserted)) {
+  if (result && Array.isArray(result?.upserted)) {
```

Comment:
no need for the `result?` if you check the existence of `result` here (also below). In fact, it looks like maybe we should just check for the existence of `result` above one time, rather than guarding it all over the place below?

Reply:
Cool yeah, I adopted this here, but below it's a bit more complex below because of the errors.